src/com/cdd/bao/template/ClipboardSchema.java

```diff
@@ -179,8 +185,33 @@ private static JSONObject formatAssignment(Schema.Assignment assn) throws JSONEx
 		}
 		json.put("values", jvalues);
 		
+		if (schvoc != null) 
+		{
+			String[] groupNest = assn.groupNest();
+			for (SchemaVocab.StoredTree tree : schvoc.getTrees()) if (assn.propURI.equals(tree.propURI) && Objects.deepEquals(groupNest, tree.groupNest))
```

Comment:
constructs like this make me wonder if `schvoc` is missing a method that would do exactly this. e.g. `getTreeForAssignment`. 
```
SchemaVocab.StoredTree tree = schvoc.getTree(assignment);
if (tree != null) json.put("tree", tree);
```
This hides the implementation detail.

Reply:
Indeed, but the main use case wraps that functionality into a different section so it's not required. Adding an extra hashmap in SchemaVocab would burden the normal functionality which never uses it, whereas this particular use case is not performance sensitive. If that balance changes, then we should add in that feature.